modules/meta.analysis/R/meta.analysis.write.model.R

```diff
@@ -26,33 +26,42 @@
 ##' @param tauB parameter b for gamma prior on precision
 ##' @return Nothing, but as a side effect, the model is written
 ##' @author David LeBauer and Mike Dietze, based on original work on the \code{write.model} function in the \code{R2WinBUGS} package by Jouni Kerman and Uwe Ligges.
-write.ma.model <- function (modelfile, outfile, reg.model, pr.dist, pr.param.a, pr.param.b, n, trt.n, site.n, ghs.n, tauA, tauB) {
-  model.text <- scan(file=modelfile, what="character",sep="@")
+write.ma.model <- function(modelfile, outfile, reg.model, pr.dist, pr.param.a, pr.param.b, n, 
+                           trt.n, site.n, ghs.n, tauA, tauB) {
+  
+  model.text <- scan(file = modelfile, what = "character", sep = "@")
   ## chose an uncommon separator in order to capture whole lines
   model.text <- gsub("%_%", "", model.text)
   model.text <- gsub("REGMODEL", reg.model, model.text)
-  model.text <- gsub("PRIORDIST", paste("d", pr.dist,sep=""), model.text)
+  model.text <- gsub("PRIORDIST", paste("d", pr.dist, sep = ""), model.text)
   model.text <- gsub("PRIORPARAMA", pr.param.a, model.text)
   model.text <- gsub("PRIORPARAMB", pr.param.b, model.text)
   model.text <- gsub("LENGTHK", n, model.text)
   model.text <- gsub("LENGTHJ", trt.n, model.text)
   model.text <- gsub("LENGTHG", site.n, model.text)
-  model.text <- gsub("TAUA", format(signif(tauA,2), scientific=FALSE), model.text)
-  model.text <- gsub("TAUB", format(signif(tauB,2), scientific=FALSE), model.text)
-  if(ghs.n == 1)  model.text <- gsub("\\#GGG", '\\#', model.text)
-  if(site.n == 1) model.text <- gsub("\\#SSS", '\\#', model.text)
-  if(trt.n == 1)  model.text <- gsub("\\#TTT", '\\#', model.text)
-  if(ghs.n > 1)   model.text <- gsub("\\#GGG", '', model.text)
-  if(site.n > 1)  model.text <- gsub("\\#SSS", '', model.text)
-  if(trt.n > 1)   model.text <- gsub("\\#TTT", '', model.text)
-  if(pr.dist == 'beta' & pr.param.b < 1) {
+  model.text <- gsub("TAUA", format(signif(tauA, 2), scientific = FALSE), model.text)
+  model.text <- gsub("TAUB", format(signif(tauB, 2), scientific = FALSE), model.text)
```

Comment:
@ashiklom It's been at least five years since I wrote that ... if it ain't broke ... but if I were going to substantially refactor this code I'd take a hard look at other packages that have been released since then and use more conventional modeling syntax.